### PR TITLE
feat: Invite flow

### PR DESCRIPTION
**Pull request by AndyLnd**

### CODE REVIEW

File: app/script/auth/module/action/InviteAction.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      /*      .then(() =>
+        dispatch(
+          TrackingAction.trackEvent({
+            action: 'invited_member',
+            category: 'team',
+            name: JSON.stringify({team_size: members && members.length}),
+          })
+        )
+      )
+*/ .catch(
```
Comment (lipis) [inline app/script/auth/module/action/InviteAction.js]: no commented out code in the codebase.. :)

delete or use it..

File: app/script/auth/page/InitialInvite.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            <ErrorMessage>{parseError(error)}</ErrorMessage>
+          </div>
+          <div>
+            {invites.length ? (
+              <ButtonLink href={nextLocation} style={{margin: '0 auto -16px'}} data-uie-name="do-next">
+                Next
+              </ButtonLink>
+            ) : (
+              <Link href={nextLocation} data-uie-name="do-skip">
+                Skip for now
```
Comment (lipis) [inline app/script/auth/page/InitialInvite.js]: `{'Skip for now'}`

File: app/script/auth/page/InitialInvite.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+                />
+                <RoundIconButton icon="plane" type="submit" data-uie-name="do-send-invite" />
+              </InputSubmitCombo>
+            </Form>
+            <ErrorMessage>{parseError(error)}</ErrorMessage>
+          </div>
+          <div>
+            {invites.length ? (
+              <ButtonLink href={nextLocation} style={{margin: '0 auto -16px'}} data-uie-name="do-next">
+                Next
```
Comment (lipis) [inline app/script/auth/page/InitialInvite.js]: `{'Next'}`

Comment (AndyLnd) [reply app/script/auth/page/InitialInvite.js]: thanks, i need to take care of translations as well

Comment (lipis) [reply app/script/auth/page/InitialInvite.js]: or that..

File: app/script/auth/module/action/SelfAction.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import * as SelfActionCreator from './creator/SelfActionCreator';
+
+export function fetchSelf() {
+  return function(dispatch, getState, {apiClient}) {
+    dispatch(SelfActionCreator.startFetchSelf());
+    return apiClient.self.api
+      .getSelf()
+      .then(selfUser => {
+        return apiClient.teams.team.api.getTeams().then(({teams}) => {
+          const boundTeams = teams.filter(team => team.binding && team.binding === true);
```
Comment (lipis) [inline app/script/auth/module/action/SelfAction.js]: is there any other way in order to avoid `=== true`?

Comment (lipis) [reply app/script/auth/module/action/SelfAction.js]: Actually we can just omit it..

```js
teams.filter(team => team.binding && team.binding)
```

File: app/script/auth/module/action/SelfAction.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+          const firstTeam = boundTeams.length ? boundTeams[0] : null;
+          if (firstTeam) {
+            selfUser.team = firstTeam.id;
+          }
```
Comment (lipis) [inline app/script/auth/module/action/SelfAction.js]:
```js
if (boundTeams.length) {
  selfUser.team = boundTeams[0].id;
}
```

File: app/script/auth/module/action/SelfAction.js (hunk 4 above)
Comment (gregor) [reply app/script/auth/module/action/SelfAction.js]: Rather `teams.filter(team => team && team.binding)`

Comment (AndyLnd) [reply app/script/auth/module/action/SelfAction.js]: copy paste shenanigans!

File: app/script/auth/module/action/InviteAction.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+import * as InviteSelector from '../selector/InviteSelector';
+import * as selfSelector from '../selector/SelfSelector';
+import * as languageSelector from '../selector/LanguageSelector';
+import BackendError from './BackendError';
+
+export function invite(invitation) {
+  const params = [...arguments];
+  console.log(invitation);
+  return function(dispatch, getState, {apiClient}) {
+    console.log('in invite');
```
Comment (gregor) [inline app/script/auth/module/action/InviteAction.js]: Remove console.logs throughout code before merging

File: app/script/auth/module/action/SelfAction.js (hunk 4 above)
Comment (lipis) [reply app/script/auth/module/action/SelfAction.js]: copy pAsta :)

File: app/i18n/webapp-cs.json [hunk 7]
```diff
@@ -16,8 +16,41 @@
...
+  "BackendError.LABEL.HANDLE_EXISTS": "",
+  "BackendError.LABEL.INVALID_HANDLE": "",
+  "BackendError.LABEL.INVALID_INVITATION_CODE": "",
+  "BackendError.LABEL.NO_OTHER_OWNER": "",
+  "BackendError.LABEL.NO_TEAM": "",
+  "BackendError.LABEL.NO_TEAM_MEMBER": "",
+  "BackendError.LABEL.TOO_MANY_MEMBERS": "",
+  "BackendError.LABEL.SUSPENDED": "Pro tento účet již není povoleno se přihlásit.",
+  "BackendError.LABEL.EMAIL_EXISTS":
+    "Tato e-mailová adresa je již používána. <a target=\"_blank\" rel=\"noopener noreferrer\" href=\"https://support.wire.com/hc/articles/115004082129\"> Další informace</a>",
```
Comment (lipis) [inline app/i18n/webapp-cs.json:54]: maybe we can increase the limit here: https://github.com/wireapp/wire-webapp/blob/dev/.prettierrc.yaml#L15

Comment (lipis) [reply app/i18n/webapp-cs.json:54]: as we did here: https://github.com/wireapp/wire-admin/blob/staging/.prettierrc.yaml#L10

Comment (AndyLnd) [reply app/i18n/webapp-cs.json:54]: I wouldn't, as there will always be lines that break that limit.

File: app/script/auth/module/action/SelfAction.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  return function(dispatch, getState, {apiClient}) {
+    dispatch(SelfActionCreator.startFetchSelf());
+    return apiClient.self.api
+      .getSelf()
+      .then(selfUser => {
+        return apiClient.teams.team.api.getTeams().then(({teams}) => {
+          const boundTeams = teams.filter(team => team.binding);
+          if (boundTeams) {
+            selfUser.team = boundTeams[0].id;
+          }
```
Comment (gregor) [inline app/script/auth/module/action/SelfAction.js:33]:
````
const [boundTeam] = teams.filter(team => team.binding);
  if (boundTeam) {
    selfUser.team = boundTeam.id;
  }
````

File: app/i18n/webapp-cs.json (hunk 7 above)
Comment (lipis) [reply app/i18n/webapp-cs.json:54]: not if we set it to `Infinity` :P

(green to merge)

**Pull request merged**